x-pack/legacy/plugins/rollup/public/crud_app/app.js

```diff
@@ -4,42 +4,34 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { HashRouter, Switch, Route, Redirect } from 'react-router-dom';
+import { HashRouter, Switch, Route, Redirect, withRouter } from 'react-router-dom';
 
 import { UIM_APP_LOAD } from '../../common';
 import { CRUD_APP_BASE_PATH } from './constants';
 import { registerRouter, setUserHasLeftApp, trackUiMetric, METRIC_TYPE } from './services';
 import { JobList, JobCreate } from './sections';
 
-class ShareRouter extends Component {
-  static contextTypes = {
-    router: PropTypes.shape({
-      history: PropTypes.shape({
-        push: PropTypes.func.isRequired,
-        createHref: PropTypes.func.isRequired
-      }).isRequired
-    }).isRequired
-  }
+const ShareRouterComponent = React.memo(({ children, history }) => {
+  useEffect(() => {
+    registerRouter({ history });
```

Comment:
When I navigate to Management > Rollup Jobs there's a Console error that indicates that `registerRouter` never gets called here. Perhaps we don't need to use the hook?

```js
const ShareRouterComponent = React.memo(({ children, history }) => {
  registerRouter({ history });

  return children;
});
```

I tried this and it seems to work.

Reply:
that would cause to call `registerRouter` on each re-render, changed back to the `Component`